src/backend/main/py.main/caliopen_main/user/core/contact.py

```diff
@@ -245,6 +255,69 @@ def lookup(cls, user, value):
         # XXX something else to do ?
         return None
 
+    @classmethod
+    def update(cls, user=None, contact_id=None, patch=None):
+        """
+        update a contact with rfc7396 Merge patch specifications
+
+        :param user: user_id
+        :param contact: contact_id
+        :param patch: dict describing the patch to apply to contact
+                with a "current_state" key. see caliopen patch rfc for explanation
+        :return: Exception or None
+        """
+        if user is None or contact_id is None or patch is None or 'current_state' not in patch:
+            return main_errors.PatchUnprocessable()
+
+        try:
+            contact = cls.get(user, contact_id)  # returns a main.user.core.contact.Contact
+        except NotFound as exc:
+            return exc
+
+        contact_store = contact.model
+        patch_current_state = patch.pop("current_state")
+
+        # populate patch to a contactParams instance to validate fields against model
+        try:
+            contact_model = params_contact.Contact(patch)
+            contact_model.validate(partial=True)
+        except Exception:
+            return main_errors.PatchUnprocessable(message="Patch does not validate against model")
+
+        # check if patch 'current_state' is consistent with current db state
+        from ..returns import ReturnContact
```

Comment:
Don't know how to compare apples and pears with current object model ! 
New object model proposal with PR #97